runtime/bcutil/cfreader.c

```diff
@@ -2833,7 +2836,7 @@ j9bcutil_readClassFileBytes(J9PortLibrary *portLib,
 	Trc_BCU_j9bcutil_readClassFileBytes_Entry();
 
 	/* There must be at least enough space for the classfile struct. */
-	if (segmentLength < (UDATA) sizeof(J9CfrClassFile)) {
+	if (segmentLength < (UDATA) sizeof(J9CfrClassFile) || segmentLength < (UDATA) sizeof(J9CfrError)) {
```

Comment:
Please update the comment to reflect the extra check. Both arguments to `||` should be parenthesized.